js/ui/date_box/ui.date_box.mask.js

```diff
@@ -325,10 +325,18 @@ const DateBoxMask = DateBoxBase.inherit({
 
     _prepareRegExpInfo() {
         this._regExpInfo = getRegExpInfo(this._getFormatPattern(), dateLocalization);
-        const regExp = this._regExpInfo.regexp;
-        const flags = regExp.flags;
-        const convertedRegExp = numberLocalization.convertDigits(this._regExpInfo.regexp.source, false);
-        this._regExpInfo.regexp = RegExp(convertedRegExp, flags);
+        const source = this._regExpInfo.regexp.source;
+        const flags = this._regExpInfo.regexp.flags;
```

Comment:
let's move `this._regExpInfo.regexp` to a temporary variable to avoid copypaste?